### PR TITLE
[Q-Page] New `styleFn` prop for QPage

### PR DESCRIPTION
**Pull request by mesqueeb**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested with all Quasar themes
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar-framework.org/tree/dev/source) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

I need this prop to be able to set **the q-page's height, in addition to the minHeight**.
I need it to be able to fix this kind of problem:

the codePen:
https://codepen.io/mesqueeb/pen/aGeKjm

Will probably also fix the carousel problem of #1902

### CODE REVIEW

**Comment by mesqueeb** [comment]
PS I will update docs as well if this would be accepted. ♡

**Comment by mesqueeb** [comment]
Update QPage.js

@rstoenescu 
I want to shortly discuss this component.
Today I had to make my app layout float in the middle of the screen when the window is high and wide.
(see screenshot below)

This made me having to override the `min-height` that's encoded on the `q-page` render. I had some trouble doing this! Trying to set `min-height` through style on the element got overridden sometimes by the quasar's style which is here.

This made me think the best way to tackle situations where the dev needs to custom manage `height`, `min-height` and `max-height`.

Instead of adding the possibility for the dev to also set the `height` through a boolean prop, I think it's better to make the developer set those things himself outside of this native component, but then give the option **to disable the `min-height` being set by quasar.**

I think this will give much more freedom.
However, the only problem in that case is that the developer would probably want to have access to the offset:
```js
const offset =
        (this.layout.header.space ? this.layout.header.size : 0) +
        (this.layout.footer.space ? this.layout.footer.size : 0)
```
to be able to keep this in mind when adding custom (min/max)height to the q-page.

So to summarize:

### my suggestion:
"I removed the `height` prop again, set a `minHeight` prop which the dev can set to `false` if he wants to prevent q-page from setting the minHeight. If it's `true` or `undefined` it will just set the minHeight like always.

### My Question:
Is it possible to access this offset somehow in vue components? If we can give the dev access to the offset in a component it's better to write in the docs on how to do that and further not set the props on this component.

PS: Good job on the Vue event today!!


image

**Comment by rstoenescu** [comment]
Ok, so best way to go is to use a `style-fn` prop, which is a function. It takes the offset as parameter, and it needs to return the style that will be applied to QPage. Available in future Quasar v0.16.1.

Would you be so kind to add this to docs? Thanks!

**Comment by mesqueeb** [comment]
@rstoenescu
Currently the the custom `styleFn` prop is available on QPage only.
Any chance we can extend this to the QLayout, `#q-app` and `body` as well?

Since all these also use `vh` which I must replace because of Mobile Safari's improper handling of `vh`.
